File: backend/src/main/java/com/slior/dto/route/CreateRouteRequest.java
package com.slior.dto.route;

import jakarta.validation.Valid;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;

import java.time.LocalDate;
import java.util.List;
import java.util.UUID;

public record CreateRouteRequest(

        @NotBlank(message = "El nombre de la ruta es obligatorio")
        String nombre,

        @NotNull(message = "La fecha planificada es obligatoria")
        LocalDate fechaPlanificada,

        @NotNull(message = "El repartidor es obligatorio")
        UUID repartidorId,

        @NotEmpty(message = "La ruta debe tener al menos una parada")
        @Valid
        List<StopRequest> paradas,

        String notas
) {}

File: mobile-app/app/src/main/java/com/slior/data/remote/dto/CreateRouteRequest.kt
package com.slior.data.remote.dto

data class StopRequestDto(
    val direccion: String,
    val destinatario: String,
    val telefonoDestinatario: String,
    val latitud: Double,
    val longitud: Double,
    val notas: String? = null
)

data class CreateRouteRequest(
    val nombre: String,
    val fechaPlanificada: String,
    val repartidorId: String,
    val paradas: List<StopRequestDto>,
    val notas: String? = null
)